tests/functional_test_base.go

```diff
@@ -437,3 +437,7 @@ func (s *FunctionalTestBase) registerArchivalNamespace(archivalNamespace string)
 	)
 	return err
 }
+
+func (s *FunctionalTestBase) overrideDynamicConfig(t *testing.T, setting dynamicconfig.GenericSetting, value any) {
```

Comment:
great, this is a lot cleaner. I think it can be improved even more to have this call s.T() instead of having the caller do it.

Reply:
Done